oauth2_http/java/com/google/auth/oauth2/AwsCredentials.java

```diff
@@ -125,6 +127,29 @@ static class AwsCredentialSource extends CredentialSource {
       } else {
         this.imdsv2SessionTokenUrl = null;
       }
+
+      this.validateMetadataServerUrls();
+    }
+
+    private void validateMetadataServerUrls() {
+      this.validateMetadataServerUrlIfAny(this.regionUrl, "region_url");
+      this.validateMetadataServerUrlIfAny(this.url, "url");
+      this.validateMetadataServerUrlIfAny(this.imdsv2SessionTokenUrl, "imdsv2_session_token_url");
+    }
+
+    private void validateMetadataServerUrlIfAny(String urlString, String nameOfData) {
```

Comment:
private **static** void validateMetadataServerUrlIfAny(String **url**, String **nameInConfig**)

Reply:
have another variable called url inside. Made the remaining changes,